File: Programming Languages/Python/0001_Two_Sum/solution_2.py
# 0001. Two Sum
# Difficulty : Easy
# Tags       : array, hash-table
# Solved on  : 2026-04-19
# Attempt    : #2
# LeetCode   : https://leetcode.com/problems/two-sum/

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        res = []
        for i in range(0,len(nums)):
            for j in range (i+1,len(nums)):
                if ( target - nums[i]) == nums[j]:
                    res.append(i)
                    res.append(j)
                    return res
        return res;
        

File: DSA/Array/0001_Two_Sum/javascript/solution_2.js
// 0001. Two Sum
// Difficulty : Easy
// Tags       : array, hash-table
// Solved on  : 2026-04-19
// Attempt    : #2
// LeetCode   : https://leetcode.com/problems/two-sum/

/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    var taken = {};
    for(var i = 0; i<nums.length;i++){
        var sub = target - nums[i];
        if (sub in taken){
            return [taken[sub],i];
        }
        taken[nums[i]] = i;
    }
};